src/demo.rs

```diff
@@ -718,22 +755,54 @@ where
                     tracing::info!("\t{}", opt);
                     values.push(opt);
                 }
-
                 self.votes.insert(options.voteidx, values);
+
+                // Replay shunted messages if we have them. This ensures that we don't print VoteCast events for Vote we haven't seen the
+                // VoteOptions event for yet.
+                if self.shunted_vote_cast_messages.is_empty() {
+                    break 'voteOptionsEvent;
+                }
+
+                // We need to temporarily move the event queue into a local buffer so we can immutably borrow self
+                // inside the closure. Once we are done, we move the queue back into self.shunted_vote_cast_messages
+                let mut temp = Vec::new();
+                std::mem::swap(&mut temp, &mut self.shunted_vote_cast_messages);
+                temp.retain(|(voteidx, event, steamid)| {
+                    // If we have a shunted message for this voteidx (because we saw the vote cast event before the vote options event)
+                    // Then retrieve it and print it now.
+                    if *voteidx == options.voteidx || self.votes.contains_key(voteidx) {
```

Comment:
I don't think this if statement is even necessary is it? Considering `get_vote_cast_message` will return `None` if this were false anyways, and I don't think it's a particularly expensive function to call (especially at the frequency it would be called anyways (that is to say, not often)).

In a similar vein, the check above to break out early isn't strictly necessary either as the vector will just be empty so nothing happens.

Just a minor preference thing you can change if you want, but not a huge deal either way.

Reply:
removing that if lets me remove a field from the shunted message tuple, so i did it. Leaving the other one that saves a mem swap